File: internal/indexer/extract/generic/testdata/cpp/sample.cpp
#include <vector>
#include "repo.hpp"

namespace store {

/// Base repository.
class Repo : public Base, private Mixin {
public:
    Repo();
    virtual ~Repo();
    virtual int find(int id) const = 0;
    static Repo* create();

protected:
    int count_;

private:
    void reset();
};

struct Point {
    int x;
    int y;
    int norm() const { return x * x + y * y; }
};

template <typename T>
T max_of(T a, T b) {
    return a > b ? a : b;
}

Repo::Repo() : count_(0) {
    reset();
}

int Repo::find(int id) const {
    auto p = new Point();
    std::sort(items.begin(), items.end());
    return helper::lookup(id);
}

enum class Mode { Read, Write };

using IdList = std::vector<int>;

}  // namespace store

int main() {
    store::Repo r;
    return r.find(1);
}

File: internal/indexer/extract/generic/testdata/rust/sample.rs
//! Crate docs.
use std::collections::HashMap;
use crate::store::{Repo, Cache as LruCache};
use super::util::*;
pub use self::model::User;
pub use crate::errors::Error as StoreError;

mod model;

/// A key-value store.
#[derive(Debug, Clone)]
pub struct Store<K> {
    pub name: String,
    items: HashMap<K, String>,
}

pub(crate) enum Mode {
    Read,
    Write(u32),
}

pub trait Backend: Send {
    fn get(&self, key: &str) -> Option<String>;
    fn put(&mut self, key: &str) {}
}

impl<K> Store<K> {
    /// Creates a store.
    pub fn new(name: &str) -> Self {
        let items = HashMap::new();
        helper(1);
        Store { name: name.to_string(), items }
    }

    async fn flush(&self) {
        self.items.clear();
    }
}

impl<K> Backend for Store<K> {
    fn get(&self, key: &str) -> Option<String> {
        util::lookup(key)
    }
}

pub const LIMIT: usize = 10;
static COUNTER: u32 = 0;
type Result<T> = std::result::Result<T, StoreError>;

macro_rules! log {
    ($x:expr) => { println!("{}", $x) };
}

#[deprecated]
fn helper(n: i32) -> i32 {
    n + 1
}

#[cfg(test)]
mod tests {
    #[test]
    fn it_works() {
        super::helper(2);
    }
}
